src/ResourceManager/DataLakeStore/Commands.DataLakeStore/DataPlaneCommands/SetAzureRmDataLakeStoreItemExpiry.cs

```diff
@@ -0,0 +1,93 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.DataLakeStore.Models;
+using Microsoft.Azure.Commands.DataLakeStore.Properties;
+using System;
+using System.Management.Automation;
+using Microsoft.Azure.DataLake.Store;
+
+namespace Microsoft.Azure.Commands.DataLakeStore
+{
+    [Cmdlet(VerbsCommon.Set, "AzureRmDataLakeStoreItemExpiry", SupportsShouldProcess = true, DefaultParameterSetName = BaseParameterSetName), OutputType(typeof(DataLakeStoreItem))]
+    [Alias("Set-AdlStoreItemExpiry")]
+    public class SetAzureRmDataLakeStoreItemExpiry : DataLakeStoreFileSystemCmdletBase
+    {
+        internal const string BaseParameterSetName = "SetAbsoluteNeverExpireExpiry";
+        internal const string RelativeExpiryParameterSetName = "SetRelativeExpiry";
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 0,ParameterSetName = BaseParameterSetName, Mandatory = true,
+            HelpMessage = "The Data Lake Store account to perform the action in")]
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 0, ParameterSetName = RelativeExpiryParameterSetName, Mandatory = true,
+            HelpMessage = "The Data Lake Store account to perform the action in")]
+        [ValidateNotNullOrEmpty]
+        [Alias("AccountName")]
+        public string Account { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 1,ParameterSetName = BaseParameterSetName, Mandatory = true,
+            HelpMessage = "The name of the firewall rule.")]
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 1, ParameterSetName = RelativeExpiryParameterSetName, Mandatory = true,
+            HelpMessage = "The name of the firewall rule.")]
+        [ValidateNotNullOrEmpty]
+        public DataLakeStorePathInstance Path { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 2, ParameterSetName = BaseParameterSetName, Mandatory = false,
+            HelpMessage =
+                "The expiration time for the specified file. If no value or set to MaxValue, the file will never expire.")]
+        [ValidateNotNullOrEmpty]
+        public DateTimeOffset? Expiration { get; set; }
```

Comment:
Why are these nullable?  No reason for the cmdlet values to be nullable and include a ValidateNotNullOrEmpty - simply make them value types, and use MyInvocation.BoundParameters to determine whether the user provided them.